src/Common/Services/InfrastructureCreationService.cs

```diff
@@ -0,0 +1,181 @@
+﻿using Cmf.CustomerPortal.BusinessObjects;
+using Cmf.CustomerPortal.Orchestration.CustomerEnvironmentManagement.InputObjects;
+using Cmf.LightBusinessObjects.Infrastructure;
+using Cmf.LightBusinessObjects.Infrastructure.Errors;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Cmf.CustomerPortal.Sdk.Common.Services
+{
+    public class InfrastructureCreationService : Utilities
+    {
+        private static TimeSpan _defaultSecondsTimeout = TimeSpan.FromSeconds(180);
+
+        /// <summary>
+        /// Check if Customer Infrastructure already exists. 
+        /// </summary>
+        /// <param name="session">Session</param>
+        /// <param name="customerPortalClient">Client to make requests to the Customer Portal's APIs</param>
+        /// <param name="ignoreIfExists">IgnoreIfExist is a variable that indicates if the program can continue if the Customer Infrastructure already exists.</param>
+        /// <param name="customerInfrastructureName">Name of Customer Infrastructure</param>
+        /// <returns></returns>
+        public static async Task<CustomerInfrastructure> CheckCustomerInfrastructureAlreadyExists(ISession session, ICustomerPortalClient customerPortalClient, bool ignoreIfExists, string customerInfrastructureName)
+        {
+            CustomerInfrastructure currentCustomerInfrastructure = null;
+            try
+            {
+                currentCustomerInfrastructure = await GetCustomerInfrastructure(session, customerPortalClient, customerInfrastructureName);
+            }
+            catch (NotFoundException) { }
+
+            // if customerInfrastructure doesn't exist. Can continue to be created.
+            // else check the ignoreIfExists value
+
+            if (currentCustomerInfrastructure != null)
+            {
+                if (ignoreIfExists)
+                {
+                    session.LogInformation($"The Customer Infrastructure with name '{customerInfrastructureName}' already exists but the 'ignoreIfExists' flag has been set. Will be used the current Customer Infrastructure.");
```

Comment:
Do not refer 'ignoreIfExists', since it's not exactly the option name neither on cmd/shell nor on powershell. Just inform that the infrastructure already exists and it will be used.

Reply:
Ok I will change it :)